pennylane/math/quantum.py

```diff
@@ -303,32 +304,32 @@ def _batched_partial_trace(density_matrix, indices):
         state_indices = "".join(state_indices)
 
         einsum_indices = f"a{state_indices}"
-        density_matrix = einsum(einsum_indices, density_matrix)
+        matrix = einsum(einsum_indices, matrix)
 
     number_wires_sub = num_indices - len(indices)
     reduced_density_matrix = np.reshape(
-        density_matrix, (batch_dim, 2**number_wires_sub, 2**number_wires_sub)
+        matrix, (batch_dim, 2**number_wires_sub, 2**number_wires_sub)
     )
     return reduced_density_matrix

```

Comment:
```
return reduced_density_matrix if batch_dim>1 else reduced_density_matrix[0]
```

Reply:
had added it but it was causing unit test failures

Reply:
added!